docs/tutorial/02-draw-a-snake.md

```diff
@@ -0,0 +1,244 @@
+
+    $ mix scenic.run
+
+Nothing changed! We need to update the configuration of our project to take this scene as the entry point for our application. So let's change the `default_scene` in the `config/config.exs` file to be our new `Game` module:
+
+```elixir
+default_scene: {Snake.Scene.Game, nil},
+```
+
+Run the application again and see there: a black screen!
+
+![sceenshot of black screen](./../images/02-black-screen.png)
+
+That's a starting point!
+
+Let's continue with the basis of our game: the pixel grid. In order to know how many squares we can fit into the scene, we need some information about the size of the application window, the so called _"viewport"_. We can retrieve that from the contextual information generated by Scenic and passed into our `init/2` function as the second `opt` parameter. We'll make use of the `info` function from Scenic's `ViewPort` module and print out what we get there:
+
+```elixir
+defmodule Snake.Scene.Game do
+  use Scenic.Scene
+
+  alias Scenic.ViewPort
+
+  def init(arg, opts) do
+    viewport = opts[:viewport]
+
+    IO.inspect(ViewPort.info(viewport), label: "viewport info")
+  end
+end
+```
+
+Running the game again, we see we have some information about the size of the view port in pixels. We can use this information to compute the number of rows and cells in our grid.
+
+When running the game you may have noticed the compiler warning us:
+
+    warning: variable "arg" is unused (if the variable is not meant to be used, prefix it with an underscore)
+
+In our configuration, we are passing `nil` as the first `arg` parameter to the `init/2` function of our `Game` scene, which already indicates that we're not planning to provide any information meant for further usage here. So let's follow the compiler's suggestions and prefix the argument with an underscore to indicate that we know about the parameter but don't need it:
+
+```elixir
+def init(_arg, opts) do
+  viewport = opts[:viewport]
+
+  IO.inspect(ViewPort.info(viewport), label: "viewport info")
+end
+```
+
+So let's use the view port information to set up our grid. We'll define the default length of 32 pixels for the edge of one square and then compute how many we can fit onto the screen. As this information is necessary throughout the whole game, we store it in the `state` of the scene.
+
+```elixir
+@tile_size 32
+
+def init(_arg, opts) do
+  viewport = opts[:viewport]
+
+  {:ok, %ViewPort.Status{size: {vp_width, vp_height}}} = ViewPort.info(viewport)
+
+  num_tiles_width = trunc(vp_width / @tile_size)
+  num_tiles_height = trunc(vp_height / @tile_size)
+
+  state = %{
+    width: num_tiles_width,
+    height: num_tiles_height
+  }
+end
+```
+
+Next we'll introduce the "graph" for our scene. The concept of a "graph" in Scenic is similar to the "DOM" in HTML: it a hierarchical description of the objects drawn onto the scene. Graphs are immutable and can only be transformed via functions. We'll set up the initial graph at compile time.
```

Comment:
Oh 🙈 will rephrase.

For the `graph` part: I adapted this from Scenic's docs. It's basically just saying that the graph is immutable. For people not familiar with the concept it may not be obvious that it can still be "updated" by functions returning a new graph. Therefore the part about functions... Will also adapt it.